src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -49,21 +51,23 @@ public KustoClient(string connectionString, string azureAccountToken, string own
 
         private GlobalState LoadSchemaState()
         {
-            CancellationTokenSource source = new CancellationTokenSource();
-
             IEnumerable<ShowDatabaseSchemaResult> tableSchemas = Enumerable.Empty<ShowDatabaseSchemaResult>();
             IEnumerable<ShowFunctionsResult> functionSchemas = Enumerable.Empty<ShowFunctionsResult>();
 
-            Parallel.Invoke(() =>
+            if (!string.IsNullOrWhiteSpace(DatabaseName))
```

Comment:
Added check to not attempt showing database information if database name is empty